appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,655 @@
+    verbose: A boolean - render all partitions if True,
+      add only three the most used partitions if False.
+
+  Returns:
+    A string with information about partition values
+      marked partition and value in red if value > 90%.
+  """
+  part_list = [[part, 100 * value["used"] / value["total"]]
+               for part, value in
+               partitions.iteritems()]  # calculate percents
+
+  # sort from more used to less used
+  part_list.sort(key=lambda p: p[1], reverse=True)
+
+  partitions_info = [
+    "'{}': {}%".format(part[0], part[1])
+    if part[1] < 90
+    else get_marked("'{}': {}%".format(part[0], part[1]), "red")
+    for part in part_list
+  ]
+
+  if not verbose:
+    if len(partitions_info) > 3:
+      partitions_info = partitions_info[:3] + ["..."]
+
+  return ", ".join(partitions_info)
+
+
+def render_memory(memory):
+  """
+  Renders memory information.
+
+  Args:
+    memory: A dict representing memory values.
+
+  Returns:
+    A string with information about node memory
+    in format: "available % / available MB".
+  """
+  return "{}% ({} MB)".format(
+    (100 * memory["available"] / memory["total"]),
+    (memory["available"] / 1024 / 1024)
+  )
+
+
+def sort_process_stats(process_stats, column, top, reverse=True):
+  """
+  Sorts process statistics by specified column.
+
+  Args:
+    process_stats: A list of process statistics.
+    column: An int representing a column number the list should be sorted by.
+    top: An int representing a process count to be printed.
+    reverse: A boolean to reverse or not reverse sorted list.
+
+  Returns:
+    A list of top processes sorted by specified column.
+  """
+  if top <= 0:
+    top = len(process_stats)
+
+  return sorted(process_stats, key=lambda p: p[column], reverse=reverse)[:top]
+
+
+def sort_proxy_stats(proxy_stats, column):
+  """
+  Sorts proxy statistics by specified column.
+
+  Args:
+    proxy_stats: A list of proxy statistics.
+    column: An int representing a column number the list should be sorted by.
+
+  Returns:
+    A list sorted by specified column.
+  """
+  return sorted(proxy_stats, key=lambda p: p[column], reverse=False)
+
+
+def get_roles(keyname):
+  """
+  Obtaines roles for each ip from AppControllerClient.
+
+  Args:
+    keyname: A string representing an identifier from AppScaleFile.
+
+  Returns:
+    A dict in which each key is an ip and value is a role list.
+  """
+  login_host = LocalState.get_login_host(keyname=keyname)
+  login_acc = AppControllerClient(
+    host=login_host,
+    secret=LocalState.get_secret_key(keyname)
+  )
+  all_private_ips = login_acc.get_all_private_ips()
+  cluster_stats = login_acc.get_cluster_stats()
+  roles_data = {
+    ip:
+      next(
+        n["roles"] for n in cluster_stats if n["private_ip"] == ip
+      )
+    for ip in all_private_ips
+  }
+
+  return roles_data
+
+
+def get_node_stats(raw_node_stats, all_roles,
+                   specified_roles=None, verbose=False):
```

Comment:
This function is called only from show_stats function and there all arguments are specified. So there is no need to assign default values to `specified_roles` and `verbose` arguments.